### PR TITLE
Preserve casing of capture groups

### PR DESCRIPTION
**Pull request by adamwroo**
-  new property `RedirectSearchData.SourceTermOriginal` added to preserve a url with original casing
-  `RedirectSearchData.SourceTermOriginal` used in `RedirectionService.GetAdditionalTargetUrlData` and `Regex.Match` with `RegexOptions.IgnoreCase` used so capture groups added to the redirect url have the original casing

Example:
As a result of this PR for the term `^test-redirect(.*)` and target `/test-target` the url:
[/test-redirect?query=AaZz123](/test-redirect?query=AaZz123)
now redirects to:
[/test-target?query=**A**a**Z**z123](/test-target?query=**A**a**Z**z123)
instead of:
[/test-target?query=**a**a**z**z123](/test-target?query=**a**a**z**z123)

### CODE REVIEW

File: be/src/Unic.UrlMapper2/code/Services/RedirectionService.cs [hunk 1]
```diff
@@ -51,16 +51,16 @@ public virtual void PerformRedirect(RedirectSearchData redirectSearchData, HttpC
...
             if (redirect is null) return;
 
-            var additionalTargetData = this.GetAdditionalTargetUrlData(redirectSearchData, redirect);
+            var additionalTargetData = this.GetAdditionalTargetUrlData(redirectSearchData.SourceTermOriginal, redirect);
 
             this.PerformRedirect(redirect, httpContext, additionalTargetData);
         }
 
-        protected virtual string GetAdditionalTargetUrlData(RedirectSearchData redirectSearchData, Redirect redirect)
+        protected virtual string GetAdditionalTargetUrlData(string sourceTerm, Redirect redirect)
```
Comment (studert) [inline be/src/Unic.UrlMapper2/code/Services/RedirectionService.cs]: **issue:** This changes the method signature of a `protected virtual` method which is intended to be an extension point. This will break backwards compatibility.
**suggestion:** Revert this change and pass the full `RedirectSearchData` object.

File: be/src/Unic.UrlMapper2/code/Services/Sanitizer.cs [hunk 2]
```diff
@@ -7,7 +7,8 @@ public class Sanitizer : ISanitizer
     {
         public virtual void SanitizeRedirectSearchData(RedirectSearchData redirectSearchData)
         {
-            redirectSearchData.SourceTerm = this.SanitizeTerm(redirectSearchData.SourceTerm);
+            redirectSearchData.SourceTerm = this.SanitizeTerm(redirectSearchData.SourceTerm)?.ToLower();
```
Comment (studert) [inline be/src/Unic.UrlMapper2/code/Services/Sanitizer.cs]: **issue:** The `ToLower()` method call was considered to be part of the sanitation, but now is hiding a bit outside of that method.
**suggestion:** Make the `SanitizeTerm` method accept an additional parameter (e.g. `convertToLower`) and give it a default value of `true` to not break backwards compatibility.

**Review by studert — CHANGES_REQUESTED** [review]

**Comment by studert** [comment]
**hint:** You also might want to update the assembly version to `1.2`

File: be/src/Unic.UrlMapper2/code/Services/RedirectionService.cs (hunk 1 above)
Comment (adamwroo) [reply be/src/Unic.UrlMapper2/code/Services/RedirectionService.cs]: good point, reverted

File: be/src/Unic.UrlMapper2/code/Services/Sanitizer.cs (hunk 2 above)
Comment (adamwroo) [reply be/src/Unic.UrlMapper2/code/Services/Sanitizer.cs]: done

**Review by studert — APPROVED** [review]
Looks good to me 👍
